src/Cli/PluginCommands.php

```diff
@@ -0,0 +1,157 @@
+        $plugin_working_copy
+        ->switchBranch($main_branch)
+        ->pull('origin', $main_branch);
+
+        // Get plugin file to parse.
+        $file_to_check = $this->getConfig()->get("plugins.$remote.file-to-check");
+        $query_string = $this->getConfig()->get("plugins.$remote.query-string");
+        $versions_file_path = "$work_dir/$file_to_check";
+        $version_file_contents = file_get_contents($versions_file_path);
+
+        // Check for updates.
+        $api_url = $this->getConfig()->get("plugins.$remote.version-api.url", '') ;
+        $latest_versions = $this->findLatestVersion($api_url);
+
+        // Get current version, compare to updated version, and update file if needed.
+        $version_file_contents = $this->getCurrentVersionUpdateFile($version_file_contents, $query_string, $latest_versions);
+
+        $message = $this->message($remote);
+        // Determine if there are any PRs already open that we should
+        // close. If its contents are the same, then we should abort rather than create the same PR again.
+        // If the contents are different, then we'll make a new PR and close this one.
+        $prs = $api->matchingPRs($plugin_working_copy->projectWithOrg(), $message);
+        if (in_array($message, $prs->titles())) {
+            $this->logger->notice("There is an existing pull request for this update; nothing else to do.");
+            return;
+        }
+        
+        file_put_contents($versions_file_path, $version_file_contents);
+
+        // Create a new pull request
+        $branch_slug = implode('-', $latest_versions);
+        $branch = $this->branchPrefix($remote) . $branch_slug;
+        $this->logger->notice('Using {branch}', ['branch' => $branch]);
+        $plugin_working_copy
+            ->createBranch($branch, $main_branch, true)
+            ->add($file_to_check)
+            ->commit($message)
+            ->push()
+            ->pr($message, '', $main_branch);
+
+        // Once we create a new PR, we can close the existing PRs.
+        $api->prClose($plugin_working_copy->org(), $plugin_working_copy->project(), $prs);
+    }
+
+    /**
+     * Get Latest Versions
+     */
+    public function findLatestVersion($api_url)
+    {
+        $availableVersions = file_get_contents($api_url);
+        if (empty($availableVersions)) {
+            throw new \Exception('Could not contact the version-check API endpoint.');
+        }
+        $versionData = json_decode($availableVersions, true);
+        if (!isset($versionData['offers'][0])) {
+            throw new \Exception('No offers returned from the version-check API endpoint.');
+        }
+        $major_version = explode('.', $versionData['offers'][0]['version']);
+        $latest_versions[] = $major_version[0] . '.' . $major_version[1];
+        $latest_versions[] = $versionData['offers'][3]['version'];
```

Comment:
Is it safe to always check for indexes 0 and 3?

Reply:
@kporras07 It's definitely always safe to check index 0 (the latest version of core), and I'm asking in WordPress core Slack about index 3. To me, the API seems to follow the pattern [on the releases page](https://wordpress.org/download/releases/), but truthfully I'm not certain on the logic. Will let you know what I learn!

Reply:
> test

Reply:
Here's the notes from WP Slack:

> You can’t really expect the order will remain the same. It’s really not designed for being used like that. The response will also change depending on the parameters passsed (and maybe even the user agent) You probably want to use https://api.wordpress.org/core/stable-check/1.0/ instead. You’d want the latest version and the highest outdated version.

> Depending on your use-case, you could also use that to tell that version 5.9.1 is deemed insecure as 5.9.2 which was a security release was released.
However, 5.9.3 might be released as not-a-security-release, and in that case, 5.9.2 would be marked as outdated and NOT insecure.

I implemented these changes in [62c929a](https://github.com/pantheon-systems/update-tool/pull/64/commits/62c929aa0bc5c6e3c218970445d5138062c56593) and things have worked well in my tests, however, I think it'd be safest to test once the next version of WP Core is released just for 100% certainty.

Reply:
Actually @kporras07, I'd be fine with merging this and then confirm/fixing after the next core release. I'll request another review.